plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -652,6 +654,25 @@ public void testOptimizingPartitionsOfV2TableWithGlobalEqualityDeleteFile()
                         .toArray(String[]::new));
     }
 
+    @Test
+    public void testOptimizingWholeTableRemovesDeleteFiles()
+            throws Exception
+    {
+        try (TestTable testTable = newTrinoTable("test_optimize_removes_obsolete_delete_files_", "AS SELECT * FROM tpch.tiny.nation")) {
+            assertUpdate("DELETE FROM " + testTable.getName() + " WHERE regionkey % 2 = 0", 15);
+            Table icebergTable = updateTableToV2(testTable.getName());
```

Comment:
How about using `loadTable` method instead? The connector creates a table with v2 by default.